From 2e5020c3913b2d4fbb7bea64462357c2c3d4e8f3 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:01 +0800
Subject: [PATCH 022/150] dt-bindings: Add bindings for Phytium HDA controller

This patch documents the DT binding for the Phytium HDA controller.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Ia826f1d04f03edc2089182bf1f4868e4820d9e65
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/sound/phytium,hda.yaml           | 51 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 52 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/phytium,hda.yaml

diff --git a/Documentation/devicetree/bindings/sound/phytium,hda.yaml b/Documentation/devicetree/bindings/sound/phytium,hda.yaml
new file mode 100644
index 000000000000..eb6394c523ec
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/phytium,hda.yaml
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/sound/phytium,hda.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium HDA controller
+
+description: |
+  The High Definition Audio (HDA) block provides a serial interface to
+  audio codec. It supports multiple input and output streams.
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    - const: phytium,hda
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    description: The interrupt from the HDA controller
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  clock-names:
+    items:
+      - const: phytium_hda_clk
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - clock-names
+
+additionalProperties: false
+
+examples:
+  - |
+    hda: hda@28206000 {
+       compatible = "phytium,hda";
+       reg = <0 0x28206000 0x0 0x1000>;
+       interrupts = <GIC_SPI 23 IRQ_TYPE_LEVEL_HIGH>;
+       clocks = <&sysclk_48mhz>;
+       clock-names = "phytium_hda_clk";
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 999ebced2fa4..a41dd0a5368b 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2533,6 +2533,7 @@ W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
 F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
+F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,qspi-nor.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,spi.yaml
 F:	arch/arm64/boot/dts/phytium/*
-- 
2.47.0

